File: 栈/LC 283移动零.java
class Solution {
    public void moveZeroes(int[] nums) {
        int stackSize = 0;
        for (int x : nums){
            if (x!=0){
                nums[stackSize++] = x;
            }
        }
        Arrays.fill(nums,stackSize,nums.length,0);
    }
}

File: 栈/LC 283移动零.cpp
//LC 283移动零 栈做法
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int stack_size = 0;
        for (int x : nums) {
            if (x) {
                nums[stack_size++] = x; // 把 x 入栈
            }
        }
        fill(nums.begin() + stack_size, nums.end(), 0);
    }
};